### ISSUE TITLE
Refactor project structure

### ISSUE THREAD

**Issue by idelsink** [derpicated/articated#11]
Refactor the project structure to represent a more professional project.
E.g. header and source files together, better CMake library division, no random `.cmake` files to define libraries, etc, etc..

**Comment by Mercotui** [comment]
With the recent addition of #9, the number of sources has grown significantly. I now firmly agree with the need for this restructure, and will therefore start implementing it.

**Comment by Mercotui** [comment]
This, db2a6670c418263668e3b8d5dfc6898e03dbb93d, is my first step in re-factoring. I moved the application source into a single dir, with subdirs for each component. The build system has been moved from `/cmake/component.cmake` files to `/component/CMakeLists.txt` files.

@idelsink I suggest you take a look at this setup, and give me feedback, as I'm planning to rework some more.
For example, one thing I'm not sure about is the header include paths, which currently follows the form of:
```cpp
augmentation_widget/augmentation_widget.hpp
```
Instead maybe the path should be provided via the build system, so that the header files can simply be included as:
```cpp
augmentation_widget.hpp
```
Not sure...
